src/coreclr/jit/morph.cpp

```diff
@@ -13160,6 +13177,50 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
 }
 #endif // defined(FEATURE_SIMD) || defined(FEATURE_HW_INTRINSICS)
 
+//------------------------------------------------------------------------
+// fgMorphModToZero: Transform a % 1 into the equivalent 0.
+//
+// Arguments:
+//    tree - The GT_MOD/GT_UMOD tree to morph
+//
+// Returns:
+//    The morphed tree, will be a GT_COMMA or a zero constant node.
+//
+GenTree* Compiler::fgMorphModToZero(GenTreeOp* tree)
+{
+    JITDUMP("\nMorphing MOD/UMOD [%06u] to Zero\n", dspTreeID(tree));
+
+    assert(tree->OperIs(GT_MOD, GT_UMOD));
+    assert(tree->gtOp2->IsIntegralConst(1));
+
+    GenTree* op1 = tree->gtGetOp1();
+    GenTree* op2 = tree->gtGetOp2();
+
+    const var_types type = tree->TypeGet();
+
+    GenTree* zero = gtNewZeroConNode(type);
+
+    if (op1->IsInvariant() || op1->IsLocal())
+    {
+        INDEBUG(zero->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);
+
+        DEBUG_DESTROY_NODE(tree->gtOp2);
+        DEBUG_DESTROY_NODE(tree->gtOp1);
+        DEBUG_DESTROY_NODE(tree);
+
+        return zero;
+    }
+
+    GenTree* comma = gtNewOperNode(GT_COMMA, tree->gtType, op1, zero);
```

Comment:
Does `op1` have to be `TYP_VOID` for this to be valid?

Reply:
I think you should use `gtExtractSideEffList` instead, that will give you a better tree to put as the LHS here (or null if there are no side effects). There are some existing examples in morph you can look at.

Reply:
Yes, this will be better since my approach was more conservative.